### PR TITLE
implemented isAppllicable and inVariant test to KickoffFriendlyPlay

### LINKED ISSUE #2257: Implement KickoffFriendlyPlay isApplicable and invariantHolds tests
### Description of the task

`isApplicable()` determines if the play should start running
`invariantHolds()` determines if the play should continue running (`isApplicable()` was true earlier to start the play)

Currently, the `isApplicable()` and `invariantHolds()` functions of the play are untested. We had a lot of bugs because of this at Virtual Robocup. We need to implement a test for each.

Take a look at an example of how to do this for `HaltPlay`, the setup of the Play, World, GameState and ASSERT_TRUE/FALSE will be similar for your test: https://github.com/UBC-Thunderbots/Software/pull/2251/files

NOTE: 
* That PR has been heavily commented to make it easier to follow along. You only need to add a couple comments to clarify what your testing, nothing more.
* The createGameState function has been implemented to conviently create GameState from referee commands. You need to poke around `software/world/game_state.cpp` to figure out the right sequence of RefereeCommands. (You should already be slightly familiar with this file from the interview :))

### Acceptance criteria

- [ ] Make sure `isApplicable` and `invariantHolds` are correct: https://robocup-ssl.github.io/ssl-rules/sslrules.html (Rule Book)
- [ ] Implement tests for `isApplicable` and `invariantHolds`

### Blocked By
N/A

### PR DESCRIPTION
**Pull request by piramonpan**
### Description

Added tests for isApplicable and invariantHold test to KickoffFriendlyPlay function

### Testing Done

Unit test 

### Resolved Issues

resolves #2257 

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [x] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [ ] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

File: src/.bazelproject [hunk 1]
```diff
@@ -18,3 +18,5 @@ additional_languages:
   # javascript
   python
   # typescript
+
+
```
Comment (akhilveeraghanta) [inline src/.bazelproject]: nit: seems to be some extra lines here

File: src/cc_toolchain/wrapper/linux_gcc [hunk 2]
```diff
@@ -6,4 +6,4 @@
 # only link in the direct dependencies for each shared lib, but because 
 # bazel delays linking to the end of the build, this means some (non-direct) 
 # dependencies of the final target will not be linked in
-exec /usr/bin/gcc -Wl,--no-as-needed "$@"
+exec /usr/bin/gcc-7 -Wl,--no-as-needed "$@"
```
Comment (akhilveeraghanta) [inline src/cc_toolchain/wrapper/linux_gcc]: Ah this was an issue from a while back, you can remove the -7, its fixed on master

**Review by akhilveeraghanta — CHANGES_REQUESTED** [review]
Nice work on the test! 

There seems to be two other files that made it in, you can undo the changes, make a new commit and push it up (git add, git commit, git push)

**Review by jonathanlew — COMMENTED** [review]
nothing to add beyond existing comments, looks good

**Review by jonathanlew — DISMISSED** [review]
lgtm pending ci

File: src/software/ai/hl/stp/play/kickoff_friendly_play_test.cpp [hunk 3]
```diff
@@ -102,4 +102,6 @@ TEST(KickoffFriendlyPlayInvariantAndIsApplicableTest, test_invariant_and_is_appl
     // Make sure we don't run the kickoff friendly play
     ASSERT_FALSE(kickoff_friendly_play.isApplicable(world));
     ASSERT_FALSE(kickoff_friendly_play.invariantHolds(world));
+
+// hi
```
Comment (LiCody) [inline src/software/ai/hl/stp/play/kickoff_friendly_play_test.cpp]: nit: we can probably remove these two lines
Also you can run ./formatting_scripts/fix_formatting,sh from the software folder to fix formatting check from failing

File: src/software/ai/hl/stp/play/kickoff_friendly_play_test.cpp [hunk 4]
```diff
@@ -71,3 +71,37 @@ TEST_F(KickoffFriendlyPlayTest, test_kickoff_friendly_play)
...
+
+    // Now lets make sure that we don't run when are NOT halted
+    world.updateGameState(::TestUtil::createGameState(
+        RefereeCommand::FORCE_START, RefereeCommand::PREPARE_KICKOFF_US));
+
+    // Make sure we don't run the kickoff friendly play
+    ASSERT_FALSE(kickoff_friendly_play.isApplicable(world));
+    ASSERT_FALSE(kickoff_friendly_play.invariantHolds(world));
+
+    // hi
```
Comment (jonathanlew) [inline src/software/ai/hl/stp/play/kickoff_friendly_play_test.cpp]: please remove this random comment

**Review by jonathanlew — COMMENTED** [review]
just a nit

**Review by jonathanlew — APPROVED** [review]
LGTM!

**Review by akhilveeraghanta — APPROVED** [review]

**Review by LiCody — APPROVED** [review]

**Pull request merged**